src/test-explorer/lcov_parser.ts

```diff
@@ -218,17 +239,19 @@ export function parseLcov(
             location = new vscode.Position(startLine, 0);
           }
           if (!info.functionsByLine.has(startLine)) {
-            // TODO: Also add demangling for C++ and Rust.
-            // https://internals.rust-lang.org/t/symbol-mangling-of-rust-vs-c/7222
-            // https://github.com/rust-lang/rustc-demangle
-            //
-            // Tested with:
-            // * Go -> no function names, only line coverage
-            // * C++ -> mangled names
-            // * Java -> mangled names
-            // * Rust -> mangled names
-            // Not tested with Python, Swift, Kotlin etc.
-            const demangled = demangleJVMMethodName(funcName) ?? funcName;
+            // Demangle the name.
+            // We must first try rustfilt before trying c++filt.
+            // The Rust name mangling scheme is intentionally compatible with
+            // C++ mangling. Hence, c++filt will be succesful on Rust's mangled
+            // names. But rustfilt provides more readable demanglings, and hence
+            // we prefer rustfilt over c++filt. For C++ mangled names, rustfilt
+            // will fail and we will fallback to c++filt.
+            // See https://internals.rust-lang.org/t/symbol-mangling-of-rust-vs-c/7222
+            const demangled =
```

Comment:
It's a shame that we don't know what language the lcov was generated from here. Is there no reasonable way of getting this info out of Bazel?

Reply:
The LCOV file from Bazel might even be merged across multiple different languages. E.g., if I run `./bazel coverage //...`, Bazel will merge coverage between Java, C++ and Go test cases.

We could use some heuristic by looking at the file extension, but then again: What would we gain by doing so?

What is your underlying concern with trying multiple decoders and using the first one that can interpret the name?

Reply:
I was just concerned that a decoder could incorrectly demangle a symbol from another language. However, I'm not sure how likely this is and if this isn't possible to easily determine the language then I'm happy with this approach.

Reply:
I think we should be good here in almost all cases. As soon as we run into it, we will find a better solution :slightly_smiling_face: